packages/app/obojobo-document-engine/src/scripts/oboeditor/components/objectives/objective-list-modal.js

```diff
@@ -0,0 +1,207 @@
+			const globalObjectives = prevState.globalObjectives.filter(
+				objective => objective.objectiveId !== objectiveID
+			)
+			return {
+				...prevState,
+				globalObjectives
+			}
+		})
+
+		this.removeObjective(objectiveID)
+
+		// // The nested loop insures that React's immutable state is updated properly
+		// return this.setState(prevState => ({
+		// 	objectivess: prevState.objectives
+		// 		.map((objective, listIndex) => (index === listIndex ? null : objective))
+		// 		.filter(Boolean)
+		// }))
+	}
+
+	createNewObjective({ label, description }) {
+		const id = uuid()
+
+		const newObjectiveObject = {
+			objectiveId: id,
+			objectiveLabel: label,
+			description
+		}
+		this.addObjective(newObjectiveObject)
+
+		this.setState(prevState => ({
+			objectives: prevState.objectives.concat({ objectiveId: id }),
+			globalObjectives: prevState.globalObjectives.concat(newObjectiveObject),
+			newObjective: false
+		}))
+	}
+
+	handleObjectiveInput(objective) {
+		this.setState({ newObjectiveInput: objective.target.value })
+	}
+
+	initializeEdit(data) {
+		this.setState({ editMode: true, editData: data })
+	}
+
+	onUpdate({ id, label, description }) {
+		// The nested loop insures that React's immutable state is updated properly
+		this.setState(prevState => {
+			return {
+				globalObjectives: prevState.globalObjectives.map(objective =>
+					objective.objectiveId === id
+						? Object.assign(objective, { objectiveLabel: label, description })
+						: objective
+				)
+			}
+		})
+		this.updateObjective(id, label, description)
+	}
+
+	onCheck(id) {
+		if (this.state.objectives.filter(o => o.objectiveId === id).length) {
+			// local objectives contain objective with objectiveId = 'id' so remove it, since deselecting it
+			this.setState(prevState => ({
+				objectives: prevState.objectives.filter(o => o.objectiveId !== id)
+			}))
+		} else {
+			// local doesn't contain objective with objectiveId = 'id' so add it
+			this.setState(prevState => ({
+				objectives: prevState.objectives.concat({ objectiveId: id })
+			}))
+		}
+	}
+
+	render() {
+		if (this.state.editMode) {
+			// Edit Dialog Popup
+			return (
+				<ObjectiveEdit
+					data={this.state.editData}
+					onCancel={() => this.setState({ editMode: false })}
+					onConfirm={state => {
+						this.onUpdate(state)
+						this.setState({ editMode: false })
+					}}
+				/>
+			)
+		}
+
+		if (this.state.newObjective) {
+			// New Objective Dialog Popup
+			return (
+				<ObjectiveInput
+					onCancel={() => this.setState({ newObjective: false })}
+					onConfirm={des => this.createNewObjective(des)}
+				/>
+			)
+		}
+
+		return (
+			<SimpleDialog
+				done
+				title="Objective Library"
+				onConfirm={() => {
+					this.props.onClose(this.state)
+				}}
+			>
+				{this.state.globalObjectives.length > 0 && this.objectiveListHeader}
+				<div
+					className="objective-list-modal"
+					style={this.state.globalObjectives.length === 0 ? { minHeight: '2.2em' } : null}
```

Comment:
Recommend adding an additional class if this condition is true, and a corresponding `.scss` rule to control this style.